File: CS/09_Charts/SetPercentageForLabels.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Presentation;
using Spire.Presentation.Charts;
using Spire.Presentation.Collections;

namespace SetPercentageForLabels
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();            
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create PPT document and load file
            Presentation ppt = new Presentation();
            ppt.LoadFromFile(@"..\..\..\..\..\..\Data\ColumnStacked.pptx");

            //Get the chart on the first slide
            IChart Chart = ppt.Slides[0].Shapes[0] as IChart;

           float dataPontPercent = 0f;

           for (int i = 0; i < Chart.Series.Count; i++)
           {
               ChartSeriesDataFormat series = Chart.Series[i];
               //Get the total number
               float total = GetTotal(series.Values);
               for (int j = 0; j < series.Values.Count; j++)
               {
                 //Get the percent
                 dataPontPercent = float.Parse(series.Values[j].Text) / total * 100;
                 //Add datalabels
                 ChartDataLabel label = series.DataLabels.Add();
                 label.LabelValueVisible = true;
                 //Set the percent text for the label
                 label.TextFrame.Paragraphs[0].Text = String.Format("{0:F2} %", dataPontPercent);
                 label.TextFrame.Paragraphs[0].TextRanges[0].FontHeight = 12;
               }
           }

           String result = "SetPercentageForLabels_result.pptx";
            //Save the document
            ppt.SaveToFile(result, FileFormat.Pptx2010);

            //Launch the result file
            PPTDocViewer(result);

        }

        private void PPTDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }        
        }

        private float GetTotal(CellRanges ranges)
        {
            float total = 0;
            for (int i = 0; i < ranges.Count; i++)
            {
                total += float.Parse(ranges[i].Text);
            }

           return total;
        }
    }
}

File: VB/09_Charts/SetPercentageForLabels.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Presentation
Imports Spire.Presentation.Charts
Imports Spire.Presentation.Collections

Namespace SetPercentageForLabels
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			'Create PPT document and load file
			Dim ppt As New Presentation()
			ppt.LoadFromFile("..\..\..\..\..\..\Data\ColumnStacked.pptx")

			'Get the chart on the first slide
			Dim Chart As IChart = TryCast(ppt.Slides(0).Shapes(0), IChart)

		   Dim dataPontPercent As Single = 0f

		   For i As Integer = 0 To Chart.Series.Count - 1
			   Dim series As ChartSeriesDataFormat = Chart.Series(i)
			   'Get the total number
			   Dim total As Single = GetTotal(series.Values)
			   For j As Integer = 0 To series.Values.Count - 1
				 'Get the percent
				 dataPontPercent = Single.Parse(series.Values(j).Text) / total * 100
				 'Add datalabels
				 Dim label As ChartDataLabel = series.DataLabels.Add()
				 label.LabelValueVisible = True
				 'Set the percent text for the label
				 label.TextFrame.Paragraphs(0).Text = String.Format("{0:F2} %", dataPontPercent)
				 label.TextFrame.Paragraphs(0).TextRanges(0).FontHeight = 12
			   Next j
		   Next i

		   Dim result As String = "SetPercentageForLabels_result.pptx"
			'Save the document
			ppt.SaveToFile(result, FileFormat.Pptx2010)

			'Launch the result file
			PPTDocViewer(result)

		End Sub

		Private Sub PPTDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Function GetTotal(ByVal ranges As CellRanges) As Single
			Dim total As Single = 0
			For i As Integer = 0 To ranges.Count - 1
				total += Single.Parse(ranges(i).Text)
			Next i

		   Return total
		End Function
	End Class
End Namespace